tests/test_spatial_methods/test_finite_volume.py

```diff
@@ -1028,6 +1029,92 @@ def test_discretise_spatial_variable(self):
             r_disc.evaluate(), 3 * disc.mesh["negative particle"][0].nodes
         )
 
+    def test_mass_matrix_Dirichlet_bcs(self):
+        """
+        Test mass matrix with Dirichlet boundary conditions
+        """
+        # one equation
+        whole_cell = ["negative electrode", "separator", "positive electrode"]
+        c = pybamm.Variable("c", domain=whole_cell)
+        N = pybamm.grad(c)
+        model = pybamm.BaseModel()
+        model.rhs = {c: pybamm.div(N)}
+        model.initial_conditions = {c: pybamm.Scalar(0)}
+        model.boundary_conditions = {
+            c: {"left": pybamm.Scalar(0), "right": pybamm.Scalar(0)}
+        }
+        model.variables = {"c": c, "N": N}
+
+        # create discretisation
+        mesh = get_mesh_for_testing()
+        spatial_methods = {"macroscale": pybamm.FiniteVolume}
+        disc = pybamm.Discretisation(mesh, spatial_methods)
+
+        combined_submesh = mesh.combine_submeshes(*whole_cell)
+        disc.process_model(model)
+
+        # mass matrix
+        mass = np.eye(combined_submesh[0].npts)
+        np.testing.assert_array_equal(mass, model.mass_matrix.entries.toarray())
+
+    def test_mass_matrix_Neumann_bcs(self):
```

Comment:
same function twice?

Reply:
Good spot, thanks! had originally had tests for Dirichlet and Neumann BCs as I thought they were handled differently. have removed the duplicate now